tests/test_kubernetes_worker.py

```diff
@@ -47,3 +47,23 @@ def test_series_upgrade(kubectl):
     assert kubectl.call_count == 2
     assert kubernetes_worker.service_pause.call_count == 2
     assert kubernetes_worker.service_resume.call_count == 2
+
+
+@patch("reactive.kubernetes_worker.is_flag_set")
+@patch("reactive.kubernetes_worker.hookenv.status_set")
+def test_status_set_on_missing_ca(mock_status_set, mock_is_flag_set):
+    """Test that set_final_status() will set blocked state if CA is missing"""
+    def is_flag_set_side_effect(flag):
+        flags = {'upgrade.series.in-progress': False,
+                 'certificates.available': False}
+        value = flags.get(flag, None)
+        assert value is not None, "Unexpected flag requested: {}".format(flag)
+        return value
+
+    mock_is_flag_set.side_effect = is_flag_set_side_effect
+
+    kubernetes_worker.charm_status()
+
+    mock_status_set.assert_called_once_with('blocked',
+                                            'Missing relation to certificate '
+                                            'authority.')
```

Comment:
Same comment re: charms.unit_test as the other PR, also fine w/ merging as-is.

```suggestion
def test_status_set_on_missing_ca():
    """Test that set_final_status() will set blocked state if CA is missing"""

    set_flag("certificates.available")
    kubernetes_worker.charm_status()
    hookenv.status_set.assert_called_with('blocked',
                                          'Connect a container runtime.')

    clear_flag("certificates.available")
    kubernetes_worker.charm_status()
    hookenv.status_set.assert_called_with('blocked',
                                          'Missing relation to certificate '
                                          'authority.')
```